lib/terraforming/resource/efs.rb

```diff
@@ -0,0 +1,62 @@
+# author: Jim Conner <snafu.x@gmail.com>
+# Nov 2016
+module Terraforming
+  module Resource
+    class EFS
+      include Terraforming::Util
+
+      def self.tf(client: Aws::EFS::Client.new)
+        self.new(client).tf
+      end
+
+      def self.tfstate(client: Aws::EFS::Client.new)
+        self.new(client).tfstate
+      end
+
+      def initialize(client)
+        @client = client
+      end
+
+      def tf
+        apply_template(@client, "tf/elastic_filesystem")
+      end
+
+      def tfstate
+        idx = -1
+
+        efsystems.inject({}) do |resources, efs|
+          idx += 1
+
+          attributes = {
+            "creation_token" => efs.creation_token,
+            "id" => efs.file_system_id,
+            "performance_mode" => efs.performance_mode,
+            "tags.%" => "1",
+            "tags.Name" => efs.name,
+          }
+
+          resources["aws_efs_file_system.efs.#{idx}"] = {
+            "type" => "aws_efs_file_system",
+            "depends_on" => [],
+            "primary" => {
+              "id" => efs.file_system_id,
+              "attributes" => attributes,
+              "meta" => {},
+              "tainted" => false,
+            },
+            "deposed" => [],
+            "provider" => "aws",
+          }
+
+          resources
+        end
+      end
+
+      private
+
+      def efsystems
```

Comment:
with respect to using an unique identifier, I originally used the file_system_id for the name. I was unclear on how that would work for the array in the case of multiple FSes. I'll use this advice for now and hopefully there will be no issues with it. All cards on the table: I'm still not a heavy user of terraform; or in other words, I'm just getting started with it. So, yeah.